src/algebra/gcd_monoid.lean

```diff
@@ -655,3 +657,34 @@ begin
     rw [pow_two, int.nat_abs_mul] at hpp,
     exact (or_self _).mp ((nat.prime.dvd_mul hp).mp hpp)}
 end
+
+instance nat.comm_cancel_monoid_with_zero : comm_cancel_monoid_with_zero ℕ :=
```

Comment:
Not sure if this is the right file for this instance. I think it can come earlier in the hierarchy.